packages/js/client-config/src/ClientConfigBuilder.ts

```diff
@@ -67,91 +62,227 @@ export class ClientConfigBuilder {
     return this.add(getDefaultClientConfig());
   }
 
-  build(): ClientConfig<Uri> {
-    this._sanitizePlugins();
-    this._sanitizeInterfacesAndImplementations();
+  addPlugin<TPluginConfig>(
+    uri: Uri | string,
+    plugin: PluginPackage<TPluginConfig>
+  ): ClientConfigBuilder {
+    const pluginUri = toUri(uri);
 
-    return this._config;
+    const existingRegistration = this._config.plugins.find((x) =>
+      Uri.equals(x.uri, pluginUri)
+    );
+
+    if (existingRegistration) {
+      existingRegistration.plugin = plugin;
+    } else {
+      this._config.plugins.push({
+        uri: pluginUri,
+        plugin: plugin,
+      });
+    }
+
+    return this;
+  }
+
+  removePlugin(uri: Uri | string): ClientConfigBuilder {
+    const pluginUri = toUri(uri);
+
+    const idx = this._config.plugins.findIndex((x) =>
+      Uri.equals(x.uri, pluginUri)
+    );
```

Comment:
Would it be possible to use `find` instead of `findIndex`? I find it more readable

Reply:
I've taken the liberty of using both across the builder. Here specifically I've decided to go with `findIndex` since it's a `remove` method, and the easiest way to remove an item from an array is `splice` which requires the `index`.
Maybe you have an alternative in mind? 😄

Reply:
Just thinking out loud, but wouldn't something like
```typescript
this._config.plugins = this._config.plugins.filter(p => !Uri.equals(p.uri, pluginUri))
```
do the job here?

Reply:
The `findIndex` and `splice` implementation is actually faster, since `filter` include redundant operations.

For example:
```typescript
const arr = ["a", "b", "c", "d"]

const findMe = "b";

// 2 iterations + 1 array shift (optimized)
const idx = arr.findIndex((x) => x === "b");
arr.splice(idx, 1);

// 4 iterations + 1 array reconstruction (not optimized)
arr.filter((x) => x === findMe);
```

It's also important to note that the overhead here isn't necessarily the iterations of the predicate (`===`), but rather the allocation of memory. In the case of `splice(..., 1)`, you can easily optimize this by doing a simple array shift. In the case of `filter((x) => ...)`, you cannot know what `...` is because it is arbitrary, so you cannot optimize the array's reconstruction as easily (not sure hot it's actually implemented).